runtime/src/main/java/org/corfudb/runtime/object/RocksDbStore.java

```diff
@@ -0,0 +1,209 @@
+package org.corfudb.runtime.object;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.collections.RocksDbEntryIterator;
+import org.corfudb.util.serializer.ISerializer;
+import org.rocksdb.BlockBasedTableConfig;
+import org.rocksdb.BloomFilter;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.Filter;
+import org.rocksdb.IndexType;
+import org.rocksdb.OptimisticTransactionDB;
+import org.rocksdb.Options;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+import org.rocksdb.WriteOptions;
+
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.locks.StampedLock;
+import java.util.stream.Collectors;
+
+/**
+ * A concrete class that implements {@link RocksDbApi} using
+ * {@link OptimisticTransactionDB}.
+ *
+ * @param <S> extends SnapshotGenerator
+ */
+@Slf4j
+public class RocksDbStore<S extends SnapshotGenerator<S>> implements
+        RocksDbApi,
+        RocksDbSnapshotGenerator<S>,
+        ColumnFamilyRegistry {
+
+    private final OptimisticTransactionDB rocksDb;
+    private final String absolutePathString;
+    private final WriteOptions writeOptions;
+    private final Options rocksDbOptions;
+
+    @Getter
+    private final ColumnFamilyHandle defaultColumnFamily;
+    @Getter
+    private final ColumnFamilyHandle secondaryIndexColumnFamily;
+
+    public RocksDbStore(@NonNull Path dataPath,
+                        @NonNull Options rocksDbOptions,
+                        @NonNull WriteOptions writeOptions) throws RocksDBException {
+        this.absolutePathString = dataPath.toFile().getAbsolutePath();
+        this.rocksDbOptions = rocksDbOptions;
+        this.writeOptions = writeOptions;
+
+        // Open the RocksDB instance
+        RocksDB.destroyDB(this.absolutePathString, this.rocksDbOptions);
+        this.rocksDb = OptimisticTransactionDB.open(rocksDbOptions, absolutePathString);
+        this.defaultColumnFamily = this.rocksDb.getDefaultColumnFamily();
+
+        // There is no need to override default options and customize
+        // the behavior of individual column families.
+        try (ColumnFamilyOptions columnFamilyOptions = new ColumnFamilyOptions()) {
+            // The prefix is composed of Index ID (1 byte) and
+            // the secondary key hash (4 bytes).
+            columnFamilyOptions.useCappedPrefixExtractor(Byte.BYTES + Integer.BYTES);
+
+            // Define prefix bloom filters, which can reduce read
+            // amplification of prefix range queries.
+            Filter bloomFilter = new BloomFilter(10);
```

Comment:
From RocksDB Tuning Guide:

**filter_policy** -- If you're doing point lookups you definitely want to turn bloom filters on. We use bloom filters to avoid unnecessary disk reads. You should set filter_policy to `rocksdb::NewBloomFilterPolicy(bits_per_key)`. Default bits_per_key is 10, which yields ~1% false positive rate. Larger bits_per_key values will reduce false positive rate, but increase memory usage and space amplification.